modules/etcd/etcd.go

```diff
@@ -0,0 +1,269 @@
+package etcd
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/testcontainers/testcontainers-go"
+	tcnetwork "github.com/testcontainers/testcontainers-go/network"
+)
+
+const (
+	clientPort          = "2379"
+	peerPort            = "2380"
+	dataDir             = "/data.etcd"
+	defaultClusterToken = "mys3cr3ttok3n"
+	scheme              = "http"
+)
+
+// EtcdContainer represents the etcd container type used in the module. It can be used to create a single-node instance or a cluster.
+// For the cluster, the first node creates the cluster and the other nodes join it as child nodes.
+type EtcdContainer struct {
+	testcontainers.Container
+	// childNodes contains the child nodes of the current node, forming a cluster
+	childNodes []*EtcdContainer
+	opts       options
+}
+
+// Terminate terminates the etcd container, its child nodes, and the network in which the cluster is running
+// to communicate between the nodes.
+func (c *EtcdContainer) Terminate(ctx context.Context) error {
+	// child nodes has no other children
+	for i, child := range c.childNodes {
+		if err := child.Terminate(ctx); err != nil {
+			return fmt.Errorf("terminate child node(%d): %w", i, err)
+		}
+	}
+
+	if err := c.Container.Terminate(ctx); err != nil {
+		return fmt.Errorf("terminate cluster node: %w", err)
+	}
+
+	// remove the cluster network if it was created, but only for the first node
+	// we could check if the current node is the first one (index 0),
+	// and/or check that there are no child nodes
+	if c.opts.clusterNetwork != nil && c.opts.currentNode == 0 {
+		if err := c.opts.clusterNetwork.Remove(ctx); err != nil {
+			return fmt.Errorf("remove cluster network: %w", err)
```

Comment:
bug: use best effort.
```suggestion
			errs = append(errs, fmt.Errorf("remove cluster network: %w", err))
```